### PR TITLE
Whole lotta genotype

### PR DESCRIPTION
**Pull request by edawson**
This isn't ready to be merged quite yet, but will bring:
     1. Updated genotype code reliant on snarls (soon)
     2. Working variant recall, outputting allele depth per each path at a variant allele stored during construction (done)
     3. Genotype likelihoods for variant recall (really soon)

### CODE REVIEW

**Comment by edawson** [comment]
This is ready to merge, though I had to revert genotype's new Snarl-based implementation as I couldn't get it to pass tests the way I'd done it. I'll do it the right way and open a fresh PR once this is in and I have time after AACR.

**Comment by ekg** [comment]
Does the snarl based implementation still produce loci calls that can be
used to edit the base graph?

**Comment by edawson** [comment]
Currently it does nothing, but I'm sure we could translate Snarls to Loci. There shouldn't be any information lost in going to Snarls.

**Comment by edawson** [comment]
@ekg @adamnovak If there are no objections I'd like to merge this in a few hours. I just don't want to break everybody's existing PR (all 11 of 'em).

File: src/subcommand/snarls_main.cpp [hunk 1]
```diff
@@ -182,6 +190,19 @@ int main_snarl(int argc, char** argv) {
...
     vector<const Snarl*> snarl_roots = snarl_manager.top_level_snarls();
+    if (fill_path_names){
+        //delete trav_finder;
+       TraversalFinder* trav_finder = new PathBasedTraversalFinder(*graph);
+        for (const Snarl* snarl : snarl_roots ){
+           vector<SnarlTraversal> travs =  trav_finder->find_traversals(*snarl);
+           stream::write_buffered(cout, travs, 0);
+        }
+
+        exit(0);
```
Comment (adamnovak) [inline src/subcommand/snarls_main.cpp]: Is this meant to be in here?

File: src/subcommand/snarls_main.cpp [hunk 2]
```diff
@@ -182,6 +190,19 @@ int main_snarl(int argc, char** argv) {
     // Load up all the snarls
     SnarlManager snarl_manager = snarl_finder->find_snarls();
     vector<const Snarl*> snarl_roots = snarl_manager.top_level_snarls();
+    if (fill_path_names){
+        //delete trav_finder;
```
Comment (adamnovak) [inline src/subcommand/snarls_main.cpp]: Do you want to actually do this delete?

File: src/vg.proto [hunk 3]
```diff
@@ -245,6 +257,7 @@ message SnarlTraversal {
 
     // A Snarl with the start and end Visits filled in to label which Snarl these traverse.
     Snarl snarl = 2;
+    string name = 3;
```
Comment (adamnovak) [inline src/vg.proto]: Are we naming the traversals individually? Or does this refer to the parent snarl?

**Comment by adamnovak** [comment]
Sounds good to me. I think whoever merges things first wins.

You might want to talk with @jeizenga eventually about the snarl naming stuff and where/when we should use that.

File: src/vg.proto (hunk 3 above)
Comment (edawson) [inline src/vg.proto]: The name is for the SnarlTraversal - each alt_path has a direct mapping to a SnarlTraversal. The Snarl names correspond to variant names.

**Comment by adamnovak** [comment]
Can you add that to the comments? Someone reading the protobuf might not
already know about this alt path naming system we have.

**Pull request merged**